static/app/views/insights/common/components/fullSpanDescription.tsx

```diff
@@ -29,16 +36,32 @@ export function FullSpanDescription({
   filters,
   moduleName,
 }: Props) {
+  const {data: indexedSpans, isFetching: areIndexedSpansLoading} = useSpansIndexed(
+    {
+      search: MutableSearch.fromQueryObject({'span.group': group}),
+      limit: 1,
+      fields: [
+        SpanIndexedField.PROJECT_ID,
+        SpanIndexedField.TRANSACTION_ID,
+        SpanIndexedField.SPAN_DESCRIPTION,
+      ],
+    },
+    'api.starfish.span-description'
+  );
+  const indexedSpan = indexedSpans?.[0];
+
+  // This is used as backup in case we don't have the necessary data available in the indexed span
   const {
     data: fullSpan,
     isLoading,
     isFetching,
-  } = useFullSpanFromTrace(group, [INDEXED_SPAN_SORT], Boolean(group), filters);
+  } = useFullSpanFromTrace(group, [INDEXED_SPAN_SORT], Boolean(indexedSpan), filters);
 
-  const description = fullSpan?.description ?? shortDescription;
+  const description =
+    indexedSpan?.['span.description'] ?? fullSpan?.description ?? shortDescription;
```

Comment:
So correct me if i'm wrong here, but were prioritizing the span description from the indexed span because were not showing large span descriptions anyways, so the truncation is okay. But if we return back a indexed span, then were still fetching the full trace to get the full span description to use as a backup?

Could we have just stayed with the original approach of only fetching the full span description, if were going to always query for it anyways? Or is there something i'm missing?